### ISSUE TITLE
[air] Ordinal Encoder complains about None

### ISSUE THREAD

**Issue by xwjiang2010** [ray-project/ray#25442]
### What happened + What you expected to happen

Should we treat them as np.NaN?

sklearn has `encoded_missing_value`

We run into this when inference data has more missing values than the training data and our Imputer doesn't account for those and Ordinal Encoder complains. 

### Versions / Dependencies

Master

### Reproduction script

NA

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
